datadriven/python/uq/refinement/RefinementStrategy.py

```diff
@@ -434,6 +437,79 @@ def update(self, grid, v, gpi, params, *args, **kws):
         # update the ranking
         return np.abs(v[ix] * fx)
 
+
+class PM1D_MC_Ranking(Ranking):
+    """
+    This is pm1d specific! DO NOT MERGE INTO MASTER!
+    Special Refinement criterion for the pm1d CO2 storage example
+    It replicates the WeightedL2OptRanking but calculates the secondMoment with Monte Carlo
+    instead of using estimated densities.
+    This is used in particular for calculating results of the pm1d code with B-splines as currently
+    there are no estimated densities available for them.
+    The Monte Carlo guess is based on 10000 sample points given by the authors of the pm1d code
+    As  the basis function integrals are independent of the actual setting they are precalculated and saved in files.
+    """
```

Comment:
This comment is a bit worrying :sweat_smile:

Reply:
Thanks for noticing. Removed